document_stores/elasticsearch/src/elasticsearch_haystack/document_store.py

```diff
@@ -0,0 +1,305 @@
+                        "Date": {
+                            "$gte": "2019-01-01"
+                        }
+                    }
+                }
+            ]
+        }
+        ```
+
+        :param filters: the filters to apply to the document list.
+        :return: a list of Documents that match the given filters.
+        """
+        query = {"bool": {"filter": _normalize_filters(filters)}} if filters else None
+
+        # Get first page of the search results
+        res = self._client.search(
+            index=self._index,
+            query=query,
+        )
+        documents = [self._deserialize_document(hit) for hit in res["hits"]["hits"]]
+
+        # This is the number of total documents that match the query
+        total = res["hits"]["total"]["value"]
+
+        # Keep querying until we have all the documents
+        from_ = len(documents)
+        while len(documents) < total:
+            res = self._client.search(
+                index=self._index,
+                query=query,
+                from_=from_,
+            )
+
+            # Add new documents to the list
+            documents.extend(self._deserialize_document(hit) for hit in res["hits"]["hits"])
+            # Update the cursor
+            from_ = len(documents)
+
+        return documents
+
+    def write_documents(self, documents: List[Document], policy: DuplicatePolicy = DuplicatePolicy.FAIL) -> None:
+        """
+        Writes (or overwrites) documents into the store.
+
+        :param documents: a list of documents.
+        :param policy: documents with the same ID count as duplicates. When duplicates are met,
+            the store can:
+             - skip: keep the existing document and ignore the new one.
+             - overwrite: remove the old document and write the new one.
+             - fail: an error is raised
+        :raises DuplicateDocumentError: Exception trigger on duplicate document if `policy=DuplicatePolicy.FAIL`
+        :return: None
+        """
+        if len(documents) > 0:
+            if not isinstance(documents[0], Document):
+                msg = "param 'documents' must contain a list of objects of type Document"
+                raise ValueError(msg)
+
+        action = "index" if policy == DuplicatePolicy.OVERWRITE else "create"
+        _, errors = helpers.bulk(
+            client=self._client,
+            actions=(
+                {
+                    "_op_type": action,
+                    "_id": doc.id,
+                    "_source": doc.to_dict(),
+                }
+                for doc in documents
+            ),
+            refresh="wait_for",
+            index=self._index,
+            raise_on_error=False,
+        )
+        if errors and policy == DuplicatePolicy.FAIL:
+            # TODO: Handle errors in a better way, we're assuming that all errors
+            # are related to duplicate documents but that could be very well be wrong.
```

Comment:
Will we address this in a future iteration?

Reply:
Yup.